### DISCUSSION TITLE
Manage Group assignment in Argocd-cm instead of using group claims in IDP integration?

### DISCUSSION THREAD

**Discussion by ArjunDandagi** [argoproj/argo-cd#13159] [Ideas]
with group claims we can get the group information of a logged in user. but for this. we need to create groups in the Identity provider and claim the groups in the token configuration. 

is it possible to just use the authentication part of the idp and then map the logged in user email to a group managed in configmap. 

# why to do this.
This will decouple the group mappings in azure . so that we don't have to manage separate groups for argocd alone. 

I am not sure if this usecase will benefit many people. but we don't want to manage infra for creating groups just for this usecase . hence asking for help 🙏

**Comment by jannfis** [comment]
You don't necessarily need to use any groups claim at all. You can configure RBAC rules for about anything in the claims, e.g. a username or an email address.

Argo CD itself has no concept of groups at all, and I don't think this will likely change.

**Reply by ArjunDandagi** [reply]
Thanks @jannfis  , the permission is little bit weird , we can't use AzureAD group name while adding roels to the groups.
 its a group id of the form
```bash
g, "84ce98d1-e359-4f3b-85af-985b458de3c6", role:org-admin 
```
how can we use groupnames instead? this hardcoding objectID seems error prone. (needs careful copy paste :p)

**Reply by jannfis** [reply]
Sorry, I'm not familiar with AzureAD at all.

You probably want to request the right OIDC scopes from Azure, and then you probably want to ask Argo CD to allow your specific scope to be used as RBAC subject (e.g. https://github.com/argoproj/argo-cd/blob/9b53eebd2856dda06685d4451a0b7316fcdd8641/docs/operator-manual/argocd-rbac-cm.yaml#L29)
